apps/heft/src/plugins/JestPlugin/jestWorkerPatch.ts

```
@@ -0,0 +1,118 @@
+// https://github.com/facebook/jest/blob/64d5983d20a628d68644a3a4cd0f510dc304805a/packages/jest-worker/src/base/BaseWorkerPool.ts#L110
+//
+//      // Schedule a force exit in case worker fails to exit gracefully so
+//      // await worker.waitForExit() never takes longer than FORCE_EXIT_DELAY
+//      let forceExited = false;
+//      const forceExitTimeout = setTimeout(() => {
+//        worker.forceExit();
+//        forceExited = true;
+//      }, FORCE_EXIT_DELAY);
+//
+// The problem is that Jest hardwires FORCE_EXIT_DELAY to be 500 ms.  On a machine that is under heavy load,
+// the IPC message is not received from the child process before the timeout elapses.  The mitigation is to
+// increase the delay.  (Jest itself seems to be a significant contributor to machine load, so perhaps reducing
+// Jest's parallelism could also help.)
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type BaseWorkerPoolModule = any;
+
+// Follow the NPM dependency chain to find the module path for BaseWorkerPool.js
+// heft --> @jest/core --> @jest/reporters --> jest-worker
+
+const PATCHED_FORCE_EXIT_DELAY: number = 7000; // milliseconds
+
+try {
+  let contextFolder: string = __dirname;
+  contextFolder = Import.resolvePackage({ packageName: '@jest/core', baseFolderPath: contextFolder });
+  contextFolder = Import.resolvePackage({ packageName: '@jest/reporters', baseFolderPath: contextFolder });
+  const jestWorkerFolder: string = Import.resolvePackage({
+    packageName: 'jest-worker',
+    baseFolderPath: contextFolder
+  });
+
+  const baseWorkerPoolPath: string = path.join(jestWorkerFolder, 'build/base/BaseWorkerPool.js');
+
+  if (!fs.existsSync(baseWorkerPoolPath)) {
+    throw new Error(
+      'The BaseWorkerPool.js file was not found in the expected location:\n' + baseWorkerPoolPath
+    );
+  }
+
+  // Load the module
+  const baseWorkerPoolModule: BaseWorkerPoolModule = require(baseWorkerPoolPath);
```

It'd be faster if you just let this throw. You could catch the error and re-throw something with a nicer message.

---

The `require()` could fail for other reasons, which are not easy to distinguish in a `catch` block